<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jean-Luc Jannink" />

<meta name="date" content="2019-10-22" />

<title>QC to GBS functions and Script</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QCtoGBSsnps</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">QC to GBS functions and Script</h1>
<h4 class="author">Jean-Luc Jannink</h4>
<h4 class="date">10/22/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>QCtoGBSsnps/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191022code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191022code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191022)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongc8c25f1"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> c8c25f1 </a>
</p>
</div>
<div id="strongRepositoryversionstrongc8c25f1" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    docs/figure/

Untracked files:
    Untracked:  data/QCAllSNPsAllPlatesJuly42018.csv
    Untracked:  data/qcSNPnamedGBSdosages.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
c8c25f1
</td>
<td>
jeanlucj
</td>
<td>
2019-10-23
</td>
<td>
First completed set of scripts to correlate QC SNPs with GBS SNPs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="convert-qc-file-to-dosages" class="section level2">
<h2>Convert QC file to dosages</h2>
<p>Function that converts nucleotide allele calls to dosages<br />
Takes a character vector where genotypes are of the form<br />
A:A, A:G, G:G<br />
All other strings in the vector will get NA dosages<br />
The QC SNP file was sent to Jean-Luc Jannink by Ismail Rabbi</p>
<pre class="r"><code>getQCdosage &lt;- function(qcVec){
  uqcVec &lt;- unique(qcVec)
  qcGen &lt;- uqcVec[nchar(uqcVec) == 3] # Genotypes have three characters
  alleles &lt;- sort(unique(unlist(strsplit(qcGen, &quot;:&quot;, fixed=T))))
  alleleCount &lt;- gregexpr(alleles[1], qcVec) # if alleles[1] absent: -1
  # if Heterozygote: length == 1, if Homozygote: length = 2
  dosage &lt;- sapply(alleleCount, length) 
  dosage[sapply(alleleCount, function(ac) ac[1] &lt; 0)] &lt;- 0 # set absent to 0
  setNA &lt;- setdiff(uqcVec, qcGen) 
  dosage[qcVec %in% setNA] &lt;- NA # Everything else gets NA
  return(list(alleles=alleles, dosage=dosage)) # Return list
}

# QC SNPs in columns, clones in rows
qcSNPs &lt;- read.csv(&quot;./data/QCAllSNPsAllPlatesJuly42018.csv&quot;, stringsAsFactors=F)
qcSNPdosage &lt;- apply(qcSNPs[,-(1:7)], 2, getQCdosage)
qcSNPalleles &lt;- sapply(qcSNPdosage, function(li) li[[1]])
qcSNPdosage &lt;- sapply(qcSNPdosage, function(li) li[[2]]) # Make dosage matrix</code></pre>
<p>Some curation on the sample names<br />
If CHECK was there, remove<br />
Some clone synonyms recorded: make them all the same</p>
<pre class="r"><code>qcSNPs$SampleNameNoChk &lt;- gsub(&quot;CHECK&quot;, &quot;&quot;, qcSNPs$SampleName)
qcSNPs$SampleNameNoChk[qcSNPs$SampleNameNoChk == &quot;I30572&quot;] &lt;- &quot;IITA-TMS-IBA30572&quot;
qcSNPs$SampleNameNoChk[qcSNPs$SampleNameNoChk %in% c(&quot;IBA980581&quot;, &quot;I980581&quot;)] &lt;- &quot;IITA-TMS-IBA3980581&quot;</code></pre>
</div>
<div id="mislabeling-in-repeat-qc" class="section level2">
<h2>Mislabeling in repeat QC</h2>
<p>Function to figure out the correlation between dosages for a clone<br />
that has been repeatedly QCed. Error threshold if correlation &lt; 0.9.<br />
Return how many mislabeled clones there were in the set<br />
Give the total number of replications and the total number of times there was<br />
mislabeling among those replications<br />
<code>colSums</code> output shows that there is a ~ 11% error rate on clones<br />
that have been nTimesQCed with QC SNPs</p>
<pre class="r"><code>nMislabeledClones &lt;- function(cloneName){
  dosages &lt;- t(qcSNPdosage[qcSNPs$SampleNameNoChk == cloneName,])
  corQC &lt;- cor(dosages, use=&quot;pairwise.complete&quot;)
  y &lt;- sum(corQC &lt; 0.9)
  n &lt;- ncol(dosages)
  # Complicated formula, simple output: given n replications
  # and y pairwise observations of lack of correlation
  # how many of the replications were mislabeled?
  return(c(n, (2*n - 1 - sqrt((2*n - 1)^2 - 4*y))/2))
}

nTimesQC &lt;- table(qcSNPs$SampleNameNoChk)
nTimesQC &lt;- nTimesQC[nTimesQC &gt; 1]

mislabelData &lt;- t(sapply(names(nTimesQC), nMislabeledClones))
rownames(mislabelData) &lt;- names(nTimesQC)
mislabelData &lt;- mislabelData[order(rownames(mislabelData)),]
colSums(mislabelData)</code></pre>
<pre><code>[1] 396  44</code></pre>
<p>For clones that have been QC’ed more than once, calculate “majority rules”<br />
allele dosages</p>
<pre class="r"><code>calcMajRuleDos &lt;- function(cloneName){
  nAndErr &lt;- mislabelData[cloneName,]
  if (nAndErr[2] &lt; nAndErr[1] / 2){ # nMislabeled less than half of replicates
    dosages &lt;- qcSNPdosage[qcSNPs$SampleNameNoChk == cloneName,]
    dosages &lt;- round(colMeans(dosages, na.rm=T))
  } else{
    dosages &lt;- rep(NA, ncol(qcSNPdosage))
  }
  return(dosages)
}
majRuleDosages &lt;- t(sapply(rownames(mislabelData), calcMajRuleDos))
majRuleDosages &lt;- majRuleDosages[!is.na(majRuleDosages[,1]),]</code></pre>
<p>Make matrices that have the same clones and SNPs, but one from GBS and<br />
the other from QC Among the GBS SNPs that I have readily available, I found 85 clones<br />
having both GBS and QC SNPs, and 37 (not 40!) SNPs in both sets</p>
<pre class="r"><code>gbsDosages &lt;- readRDS(&quot;./data/qcSNPnamedGBSdosages.rds&quot;)
# Which clones have both QC and gbs SNPs?
clonesBoth &lt;- intersect(rownames(majRuleDosages), rownames(gbsDosages))
# Which SNPS are in both QC and GBS sets?
snpsBoth &lt;- intersect(colnames(majRuleDosages), colnames(gbsDosages))
gbsOfQC &lt;- gbsDosages[clonesBoth, snpsBoth]
qcOfGBS &lt;- majRuleDosages[clonesBoth, snpsBoth]</code></pre>
</div>
<div id="common-coding-for-qc-and-gbs" class="section level2">
<h2>Common coding for QC and GBS</h2>
<p>Problem: we have dosages for GBS and dosages for the QC SNPs but we don’t<br />
know if they were <em>coded</em> the same way: maybe one allele was<br />
counted (0, 1, 2) for GBS but the other allele was counted for QC<br />
Solution: correlate the dosages across GBS and QC. If the correlation<br />
is negative, that means that they were coded with opposite alleles.<br />
If the correlation is positive, they were coded the same way.<br />
Since <code>getQCdosage</code> arbitrarily coded the SNPs one way, about half<br />
are reversed relative to GBS</p>
<pre class="r"><code>corBySNP &lt;- sapply(1:ncol(gbsOfQC), function(i) cor(gbsOfQC[,i], qcOfGBS[,i], use=&quot;pairwise.complete.obs&quot;))
hist(corBySNP)</code></pre>
<p><img src="figure/RelateQCsnpsToGBSsnps.Rmd/Correlation%20by%20SNP-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>For the SNPs that were coded opposite, flip the dosages</p>
<pre class="r"><code>qcOfGBS[,corBySNP &lt; 0] &lt;- 2 - qcOfGBS[,corBySNP &lt; 0]</code></pre>
</div>
<div id="mislabeling-between-qc-and-gbs" class="section level2">
<h2>Mislabeling between QC and GBS</h2>
<p>Now that the SNPs are all coded the same way, correlate by clone<br />
If there is not a high correlation, the GBS sample and the QC sample<br />
were not taken from the same genetic individual</p>
<pre class="r"><code>corByClone &lt;- sapply(1:nrow(gbsOfQC), function(i) cor(gbsOfQC[i,], qcOfGBS[i,], use=&quot;pairwise.complete.obs&quot;))
names(corByClone) &lt;- rownames(qcOfGBS)
# Make a pretty histogram
hist(corByClone, xlab=&quot;Correlation of dosages by GBS vs QC&quot;, main=paste0(nrow(qcOfGBS), &quot; Clones Matched&quot;), cex.axis=1.3, cex.lab=1.3)</code></pre>
<p><img src="figure/RelateQCsnpsToGBSsnps.Rmd/Find%20mislabeled%20clones-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(rownames(qcOfGBS)[corByClone &lt; 0.75])</code></pre>
<pre><code>[1] &quot;ANG18&quot;              &quot;IITA-TMS-IBA070536&quot; &quot;TMEB1&quot;             
[4] &quot;TMEB102&quot;            &quot;TMEB1417&quot;           &quot;TMEB1593&quot;          
[7] &quot;TMEB600&quot;           </code></pre>
<p>Remove clones from the set that appear to be mislabeled</p>
<pre class="r"><code>gbsOfQCClean &lt;- gbsOfQC[corByClone &gt; 0.75,]
qcOfGBSClean &lt;- qcOfGBS[corByClone &gt; 0.75,]</code></pre>
</div>
<div id="snps-where-gbs-and-qc-disagree" class="section level2">
<h2>SNPs where GBS and QC disagree</h2>
<p>With this cleaned set check the correlations between GBS and QC SNPs<br />
for each <em>SNP</em>. Low correlation means the SNP does not track well between<br />
GBS and KASP</p>
<pre class="r"><code>corBySNP &lt;- sapply(1:ncol(gbsOfQCClean), function(i) cor(gbsOfQCClean[,i], qcOfGBSClean[,i], use=&quot;pairwise.complete.obs&quot;))
hist(corBySNP)</code></pre>
<p><img src="figure/RelateQCsnpsToGBSsnps.Rmd/Remove%20SNPs%20that%20do%20not%20track-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The two worst SNPs (cor of 0.71 and 0.81) are both on Chr. 5<br />
S5_1476447 and S5_25731301</p>
<pre class="r"><code>colnames(qcOfGBSClean)[corBySNP &lt; 0.85]</code></pre>
<pre><code>[1] &quot;S5_1476447&quot;  &quot;S5_25731301&quot;</code></pre>
<pre class="r"><code>qcOfGBSClean &lt;- qcOfGBSClean[,corBySNP &gt; 0.85]
gbsOfQCClean &lt;- gbsOfQCClean[,corBySNP &gt; 0.85]</code></pre>
</div>
<div id="clones-where-qc-and-gbs-agree" class="section level2">
<h2>Clones where QC and GBS agree</h2>
<p>Retest the clones. The GBS to QC correlation per clone is improved<br />
The GBS to QC correlation per clone is improved with only good SNPs</p>
<pre class="r"><code>goodClones &lt;- sapply(1:nrow(qcOfGBSClean), function(i) cor(gbsOfQCClean[i,], qcOfGBSClean[i,], use=&quot;pairwise.complete.obs&quot;))
plot(corByClone[rownames(qcOfGBSClean)], goodClones)
lines(c(0,1), c(0,1))</code></pre>
<p><img src="figure/RelateQCsnpsToGBSsnps.Rmd/Final%20QC%20to%20GBS%20correlation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(goodClones, xlab=&quot;Correlation of dosages by GBS vs QC&quot;, main=paste0(nrow(qcOfGBSClean), &quot; Good Clones&quot;), cex.axis=1.3, cex.lab=1.3)</code></pre>
<p><img src="figure/RelateQCsnpsToGBSsnps.Rmd/Final%20QC%20to%20GBS%20correlation-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.4.0 Rcpp_1.0.2      digest_0.6.20   rprojroot_1.3-2
 [5] backports_1.1.4 git2r_0.26.1    magrittr_1.5    evaluate_0.14  
 [9] stringi_1.4.3   fs_1.3.1        whisker_0.4     rmarkdown_1.16 
[13] tools_3.6.1     stringr_1.4.0   glue_1.3.1      xfun_0.8       
[17] yaml_2.2.0      compiler_3.6.1  htmltools_0.3.6 knitr_1.24     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
